logger/include/logger.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <filesystem>
+#include <iostream>
+#include <list>
+#include <source_location>
+#include <sstream>
+
+enum Loglevel
+{
+    INFO,
+    CRITICAL
+};
+
+namespace Logger
+{
+
+std::list<std::string> buffer;
+
+// The size of buffer is 8k
+const size_t bufferSize = 8000;
+
+/**
+ *@brief Appends data to the buffer
+ *
+ *This API is called to log message to the buffer.
+ *It checks the log level and appends information like
+ *line number, file name, function name to the buffer.
+ *
+ *@param[in] level - loglevel of the message
+ *@param[in] message - Information to be logged
+ *@param[in] location - object of the source_location class
+ */
+void logMessage(
+    const Loglevel level, const std::string& message,
+    const std::source_location& location = std::source_location::current())
+{
+    if (Loglevel::INFO == level)
+    {
+        buffer.push_back(
+            " line no: " + std::to_string(location.line()) + " File: " +
```

Comment:
we need to use time stamp as well right ?

log with time-stamp looks like: 
2017-06-06:17:07:02,158 DEBUG    [log.py:11] This is a debug log
2023-11-10 15:30:45 [INFO] my_module: example.py:42 - This is an informational message.
2023-11-10 12:00:00 [INFO] [example.py:23] - Program started.